branchpro/models.py

```diff
@@ -42,3 +43,117 @@ def simulate(self, parameters, times):
             value appears in ``times``.
         """
         raise NotImplementedError
+
+#
+# BranchProModel Class
+#
+
+
+class BranchProModel(ForwardModel):
+    r"""BranchProModel Class:
+    Class for the models following a Branching Processes behaviour.
+    It inherits from the `ForwardModel`` class.
+
+    In the branching process model, we track the number of cases
+    registered each day, I_t, also known as the "incidence" at time t.
+
+    The incidence at time t is a random variable distributed according to
+    a Poisson distribution with a mean that depends on previous number of
+    cases, according to the following formula:
+
+    .. math::
+        E(I_{t}^{\text(local)}|I_0, I_1, \dots I_{t-1}, w_{s}, R_{t}) =
+            R_{t}\sum_{s=1}^{t}I_{t-s}w_{s}
+
+    Parameters
+    ----------
+    initial_r
+        (numeric) Value of the reproduction number at the beginning
+        of the epidemic
+    serial_interval:
+        (list) Unnormalised probability distribution of that the recipient
+        first displays symptoms s days after the infector first displays
+        symptoms
+
+    Methods
+    -------
+    simulate: return model output for specified parameters and times.
+    _normalised_daily_mean: (Private) returns the expected number of new cases
+        at time t
+    """
+
+    def __init__(self, initial_r, serial_interval):
+        super(BranchProModel, self).__init__()
+
+        if not isinstance(serial_interval, list):
```

Comment:
I just made a bunch of comments in #31 but I think this should be modified to allow a np.array to be passed, not just a list.